JavaScript/JavaScript-Classes.md

```diff
@@ -0,0 +1,193 @@
+
+	var House = function(colour){
+    var obj = Object.create(House.prototype); // generate object
+
+    obj.colour = colour;
+    obj.door = 'open'; //assign properties
+
+    return obj;
+	};
+
+	//(Automatically generated by interpreter)
+	//House.prototype = {};
+
+	House.prototype.openDoor = function(){
+    this.door = 'open';
+	};
+
+	House.prototype.closeDoor = function(){
+    this.door = 'close';
+	};
+
+	var house = House('red'); //instantiation pattern
+
+
+###Pseudoclassical
+
+Lastly we come to pseudoclassical. Instead of assigning the `Object.create(House.prototype)` to a new variable, it is assigned to `this` for the purpose of simple property assignment and method creation. The interpreter will do this automatically 'under the bonnet', and return the object, so long as the keyword `new` is initialised at instantiation `var house = new House('red')`. The disadvantage is that object creation is not that clear.
+
+	var House = function(colour){
+    //(Automatically generated by interpreter)
+    //var this = Object.create(House.prototype);
+    this.colour = colour;
+    this.door = 'open';
+
+    //(Automatically generated by interpreter)
+    //return this;
+	};
+
+	//(Automatically generated by interpreter)
+	//House.property = {}
+
+	House.prototype.openDoor = function(){
+    this.door = 'open';
+	};
+
+	House.prototype.closeDoor = function(){
+    this.door = 'closed';
+	};
+
+	var house = new House('red'); //instantiation pattern
+
+
+Instantiation happens when a functional class is utilized to create a new object,
+
+	var house = House('red')
+
+Note that pseudoclassical is the only style that makes use of the `new` keyword, the other three classes are used like a function call.
+
+###Summary
+
+None of these four are better or worse than each other to use, they are just styles. It is up to you as the programmer to decide which your program would benefit the most from.
+
+*Functional*
+
+* Most transparent and easy to understand
+ 
+* Higher memory cost because each instance retains its own properties
+
+*Functional-shared*
+
+* Offers memory cost advantage over functional
+
+* A bit more complex than functional
+
+*Prototypal*
+
+* Shared properties in a separate object and not extended within the instatiated object
+
+* Code is more complex to read
+
+*Pseudoclassical*
+
+* Often the most optimized pattern and very common
+
+* *this* can be difficult to understand
```

Comment:
Maybe say 'keyword this' to avoid someone thinking that 'this' refers to the pseudoclassical style. (You have it in italics, but it would be even clearer like that.)